### PR TITLE
Use AmbiguousEJBReferenceException with legacy EJB bindings

### PR DESCRIPTION
**Pull request by brideck**
- Expose com.ibm.websphere.ejbcontainer as internal IBM API
- Remove unused entity EJB APIs
- Modify legacy bindings support to throw AmbiguousEJBReferenceException
- Update bindings FAT to use AmbiguousEJBReferenceException
- Cleanup old apps in setup for each FAT test

### CODE REVIEW

**Comment by brideck** [comment]
#build

**Comment by brideck** [comment]
for #11441

**Comment by brideck** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/LocalColonEJBNamingHelperImpl.java [hunk 1]
```diff
@@ -70,25 +74,42 @@ protected void unsetEJBHomeRuntime(ServiceReference<EJBHomeRuntime> ref) {
...
-            //TODO: If LocalColonEJBBindings already contains name, bind ambiguous reference exception
-            localColonEJBBindings.put(name, binding);
+            localColonEJBBindings.put(name, localBinding);
         } finally {
             writeLock.unlock();
         }
 
         if (isTraceOn && tc.isEntryEnabled()) {
-            Tr.exit(tc, "bind");
+            Tr.exit(tc, "bind: " + notAmbiguous);
```
Comment (olendvcook) [inline dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/LocalColonEJBNamingHelperImpl.java:109]: Same here, if I saw just this in trace `bind: false` I would think the bind failed, not that it's not Ambiguous so I'd make this clearer.

**Review by tkburroughs — APPROVED** [review]

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/bnd.bnd [hunk 2]
```diff
@@ -54,4 +54,6 @@ tested.features: \
 	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
 	com.ibm.websphere.javaee.transaction.1.1;version=latest,\
 	com.ibm.websphere.security;version=latest,\
-	com.ibm.ws.ejbcontainer.fat_tools;version=latest
\ No newline at end of file
+	com.ibm.ws.ejbcontainer.core;version=latest,\
```
[the remark this answers is not in the document]
Comment (brideck) [reply dev/com.ibm.ws.ejbcontainer.bindings_fat/bnd.bnd:57]: I can give that a try.

File: dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/EJBLocalNamingHelperImpl.java [hunk 3]
```diff
@@ -81,18 +87,33 @@ public synchronized void bind(EJBBinding binding, String name) {
...
-            //TODO: If EJBLocalBindings already contains name, bind ambiguous reference exception
-            EJBLocalBindings.put(name, binding);
+            EJBLocalBindings.put(name, localBinding);
         } finally {
             writeLock.unlock();
         }
 
         if (isTraceOn && tc.isEntryEnabled()) {
-            Tr.exit(tc, "bind");
+            Tr.exit(tc, "bind: " + notAmbiguous);
```
[the remark this answers is not in the document]
Comment (brideck) [reply dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/EJBLocalNamingHelperImpl.java:113]: Updated exit message to "bind: notAmbiguous = "

File: dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/LocalColonEJBNamingHelperImpl.java (hunk 1 above)
Comment (brideck) [reply dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/naming/LocalColonEJBNamingHelperImpl.java:109]: Updated exit message to "bind: notAmbiguous = "

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/bnd.bnd (hunk 2 above)
[the remark this answers is not in the document]
Comment (brideck) [reply dev/com.ibm.ws.ejbcontainer.bindings_fat/bnd.bnd:57]: Looks like that worked fine.  I removed the remote.portable.core dependency, too.

**Comment by brideck** [comment]
#build

**Review by olendvcook — APPROVED** [review]

**Comment by brideck** [comment]
#build

**Comment by brideck** [comment]
#build

**Comment by olendvcook** [comment]
#build

**Comment by brideck** [comment]
#run-libby-bot

**Comment by olendvcook** [comment]
With the last 2 builds I feel like they are good enough to safely deliver

**Pull request merged**